shuup/front/utils/sorts_and_filters.py

```diff
@@ -0,0 +1,214 @@
+        may call `~Django.forms.Form.add_error` to add errors to form or
+        modify the ``form.cleaned_data`` dictionary.
+
+        :param form: Form that is currently cleaned
+        :type form: ProductListForm
+        :rtype: None
+        """
+        pass
+
+
+class ProductListForm(forms.Form):
+    def __init__(self, shop, category, *args, **kwargs):
+        super(ProductListForm, self).__init__(*args, **kwargs)
+        for extend_class in _get_active_modifiers(shop, category):
+            for field_key, field in extend_class().get_fields(category) or []:
+                if field_key not in self.fields:
+                    self.fields[field_key] = field
+
+            for field_key, choices in extend_class().get_choices_for_fields() or []:
+                if field_key in self.fields:
+                    self.fields[field_key].widget.choices += choices
+
+    def clean(self):
+        cleaned_data = super(ProductListForm, self).clean()
+        for extend_class in get_provide_objects(FORM_MODIFIER_PROVIDER_KEY):
+            extend_class().clean_hook(self)
+        return cleaned_data
+
+
+def get_configuration(shop=None, category=None):
+    default_configuration = configuration.get(
+        shop, FACETED_DEFAULT_CONF_KEY, settings.SHUUP_FRONT_DEFAULT_FACET_CONFIGURATION)
+    return (configuration.get(None, _get_category_configuration_key(category)) or default_configuration)
+
+
+def set_configuration(shop=None, category=None, data=None):
+    if category and category.pk:
+        configuration.set(None, _get_category_configuration_key(category), data)
+    elif shop:
+        configuration.set(shop, FACETED_DEFAULT_CONF_KEY, data)
+
+
+def get_query_filters(request, category, data):
+    filter_q = Q()
+    for extend_class in _get_active_modifiers(request.shop, category):
+        extra_filter = extend_class().get_filters(request, data)
+        if extra_filter:
+            filter_q &= extra_filter
+    return filter_q
+
+
+def post_filter_products(request, category, products, data):
+    for extend_class in _get_active_modifiers(request.shop, category):
+        products = extend_class().filter_products(request, products, data)
+    return products
+
+
+def sort_products(request, category, products, data):
+    sort = data.get("sort", "name_a")
+    for extend_class in _get_active_modifiers(request.shop, category):
+        products = extend_class().sort_products(request, products, sort)
+    return products
+
+
+def _get_category_configuration_key(category):
+    return (FACETED_CATEGORY_CONF_KEY_PREFIX % category.pk if category and category.pk else None)
+
+
+def _get_active_modifiers(shop=None, category=None):
+    configuration = get_configuration(shop=shop, category=category)
+
+    def sorter(extend_class):
+        return extend_class().get_ordering(configuration)
+
+    return sorted(
+        [c for c in get_provide_objects(FORM_MODIFIER_PROVIDER_KEY) if c().should_use(configuration)],
```

Comment:
This could return objects since you are using this everywhere without any initial params. 
```python
return sorted(
         [c() for c in get_provide_objects(FORM_MODIFIER_PROVIDER_KEY) if c().should_use(configuration)],
         key=sorter
)
```
This has, however, a double instantiation so it could be just:
```python
objs = []
for cls in get_provide_objects(FORM_MODIFIER_PROVIDER_KEY):
	obj = cls()
	if obj.should_use(configuration):
		objs.append(obj)
return sorted(objs, key=sorter)
```
or something. I also think that it's common practice to return objects from these kinds of functions instead of classes everywhere else in Shuup.